### PR TITLE
Migrate crm_action to v10

### PR DESCRIPTION
**Pull request by qtheuret**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -11,6 +11,7 @@ Available addons
 ----------------
 addon | version | summary
 --- | --- | ---
+[crm_action](crm_action/) | 10.0.1.0.0 | CRM Action
```
Comment (pedrobaeza) [inline README.md]: This is changed automatically, so please don't change it by yourself or it will conflict.

**Review by pedrobaeza — APPROVED** [review]
Only little comment about README.md (that you should squash), but overall LGTM code review

File: crm_action/README.rst [hunk 2]
```diff
@@ -1,54 +1,85 @@
...
+#. when the action is done, click on the button *Mark as done*.
+
+You can overview all actions for any lead or opportunity with the *Actions*
+menu entry. On the form view of an opportunity, you can see the next action to
+do and there is also a button to mark it as done (it will immediately display
+the new next action to do).
+
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+   :alt: Try me on Runbot
+   :target: https://runbot.odoo-community.org/runbot/111/9.0
```
Comment (cubells) [inline crm_action/README.rst]: It must be 10.0

File: crm_action/data/email_reminder.xml [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+            <field name="interval_number">1</field>
+            <field name="interval_type">days</field>
+            <field name="numbercall">-1</field> <!-- don't limit the number of calls -->
+            <field name="doall" eval="False"/>
+            <field name="model" eval="'crm.action'"/>
+            <field name="function" eval="'_send_email_reminder'" />
+            <field name="args" eval="'()'"/>
+        </record>
+    </data>
+
```
Comment (cubells) [inline crm_action/data/email_reminder.xml:22]: You must add data tag here for noupdate="0"

File: crm_action/data/email_reminder.xml [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+% endfor
+</ul>
+
+<p>
+-- <br/>
+Automatic e-mail sent by Odoo. Do not reply.
+</p>
+</div>
+]]></field>
+        </record>
```
Comment (cubells) [inline crm_action/data/email_reminder.xml:71]: You must add /data tag

File: crm_action/views/crm_action_view.xml [hunk 5]
```diff
@@ -1,80 +1,99 @@
...
+        <field name="model">crm.action</field>
+        <field name="arch" type="xml">
+            <tree editable="top" colors="red:date &lt; current_date and state=='draft';blue:date &gt;= current_date and state=='draft'">
+                <field name="lead_id"/>
+                <field name="partner_id"/>
+                <field name="date"/>
+                <field name="user_id"/>
+                <field name="action_type_id"/>
+                <field name="details"/>
+                <button icon="STOCK_APPLY" name="button_confirm" type="object" string="Mark as Done" states="draft"/>
```
Comment (cubells) [inline crm_action/views/crm_action_view.xml]: You must change gtk icons by font awesome icons

File: crm_action/views/crm_action_view.xml [hunk 6]
```diff
@@ -1,80 +1,99 @@
...
+                <button icon="gtk-undo" name="button_set_to_draft" type="object" string="Back to Todo" states="done"/>
```
Comment (cubells) [inline crm_action/views/crm_action_view.xml]: You must change gtk icons by font awesome icons

File: crm_action/models/crm_action.py [hunk 7]
```diff
@@ -75,3 +85,43 @@ def button_confirm(self):
...
+        actions = self.env['crm.action'].search([
+            ('state', '=', 'draft'), ('date', '=', today)])
+        logger.info(
+            'Preparing CRM actions email reminders '
+            '(%d actions found for today)',
+            len(actions))
+        user_company_actions = {}
+        # key = (user, company)
+        # value = list of crm.action records
+        if actions:
```
Comment (cubells) [inline crm_action/models/crm_action.py]: If action is not needed here if you have a for after.

**Review by cubells — CHANGES_REQUESTED** [review]
Fast code review

**Comment by rafaelbn** [comment]
Hi @qtheuret thanks for this contribution, please could you attend comments?

**Comment by qtheuret** [comment]
@rafaelbn Sorry to be late on this request. It should be good now.

File: crm_action/models/crm_lead.py [hunk 8]
```diff
@@ -1,49 +1,61 @@
...
 class CrmLead(models.Model):
     _inherit = 'crm.lead'
 
-    def count_actions(self):
+    def compute_count_actions(self):
         self.actions_count = len(self.action_ids)
 
-    actions_count = fields.Integer(compute='count_actions')
+    actions_count = fields.Integer(
+        compute='compute_count_actions',
```
Comment (cubells) [inline crm_action/models/crm_lead.py]: Compute method should start with _compute

Change to _compute_count_actions

File: crm_action/models/crm_lead.py [hunk 9]
```diff
@@ -1,49 +1,61 @@
...
+    )
+    title_action = fields.Char(
+        related='next_action_id.display_name',
+        readonly=True,
+        store=True,
+    )
+    next_action_id = fields.Many2one(
+        comodel_name='crm.action',
+        string='Next Action',
+        compute='compute_next_action',
```
Comment (cubells) [inline crm_action/models/crm_lead.py]: Compute method should start with _compute

Change to _compute_next_action

File: crm_action/models/crm_action.py [hunk 10]
```diff
@@ -1,72 +1,82 @@
...
+            ('draft', 'Todo'),
+            ('done', 'Done'),
+        ],
+        string='Status',
+        required=True,
+        readonly=True,
+        default="draft",
+    )
+    display_name = fields.Char(
+        compute='compute_display_name',
```
Comment (cubells) [inline crm_action/models/crm_action.py]: Compute method should start with _compute

Change to _compute_display_name

**Review by cubells — CHANGES_REQUESTED** [review]

**Review by cubells — APPROVED** [review]
:+1: 

LGTM

**Comment by qtheuret** [comment]
Can anyone tell me what's wrong with the ci/runbot check ?

**Comment by cubells** [comment]
@qtheuret 
I think It's correct. Runbot log shows a success test summary.

I reviewed warning in runbot logs and your code seems to be ok.

**Comment by rafaelbn** [comment]
@qtheuret maybe is the warning?

> 2017-08-22 20:39:35,492 186 WARNING openerp_test odoo.fields: Field crm.action.display_name depends on itself; please fix its decorator @api.depends().

We cannot merge this PR as it is, it will show the warning to the full CRM branch

**Comment by rafaelbn** [comment]
Who is the runbot master? 😄

**Comment by qtheuret** [comment]
@rafaelbn I didn't find any piece of code related to crm.action.display_name. Can you give me advice to fix this warning ?

**Comment by rafaelbn** [comment]
Please @Yajo @lasley @gurneyalex do you know how to solve runbot ✖️ ? Or any other runbot master...

**Comment by yajo** [comment]
This PR has conflicts, you have to fix those.

Besides, when you go to the runbot, you can access the logs like here:
captura de pantalla de 2017-08-30 09-47-44

In *full all logs* I found several of:

```
WARNING openerp_test odoo.fields: Field crm.action.display_name depends on itself; please fix its decorator @api.depends().
```

Most likely, that's the bug. Warnings count as errors AFAIK.

**Comment by lasley** [comment]
> Most likely, that's the bug. Warnings count as errors AFAIK.
 
Yup this is the big difference between Travis and Runbot. In the event of warnings but no error, the Runbot build card will be yellow and the pull request is marked as a fail.

**Comment by rafaelbn** [comment]
Hi @qtheuret I wish @Yajo has clarified you what you must fix 😄

**Comment by alexis-via** [comment]
I fixed the warning in odoo's log in this PR to @qtheuret 's branch https://github.com/qtheuret/crm/pull/2

**Comment by yajo** [comment]
Please rebase and fix conflicts to merge.

**Pull request closed without merging**